uast/utils.go

```diff
@@ -18,7 +18,12 @@ func Tokens(n *Node) []string {
 	return tokens
 }
 
-// CyclomaticComplexity returns the cyclomatic complexity for the node. This uses the method of
+// CyclomaticComplexity returns the cyclomatic complexity for the node. The cyclomatic complexity
+// is a quantitative measure of the number of linearly independent paths through a program's source code.
+// It was developed by Thomas J. McCabe, Sr. in 1976. For a formal description see:
+// https://en.wikipedia.org/wiki/Cyclomatic_complexity
+// And the original paper: http://www.literateprogramming.com/mccabe.pdf
+// This implementation uses the method of
```

Comment:
I'd be happier with a rationale about what Cyclomatic Complexity definition we chose and why:
- What Cyclomatic Complexity definition are we based on (I understood that we weren't using McCabe's paper definition, maybe I understood it wrong).
- Why we have selected this particular set of Roles
- Why we have decided to only increment by one with boolean operators